cleverhans/utils.py

```diff
@@ -220,6 +220,101 @@ def grid_visual(data):
     return figure
 
 
+def get_logits_over_interval(sess, model, x, adv_x, x_data,
+                             min_epsilon=-10, max_epsilon=10,
+                             num_points=21):
+    """Get logits when the input is perturbed in an interval in adv direction.
+
+    Args:
+        sess: Tf session
+        model: Model for which we wish to get logits
+        x: Data tensor of shape [None, height, width, channels]
```

Comment:
If the user has to pass in `model`, why not just make your own `x`? It really only makes sense to pass in `x` if the user also has to pass in the logits corresponding to `x`.

Reply:
We probably don't want it to be shape `None` because you're relying on the user passing in exactly one example. To enforce that, it should be shape 1.

Reply:
Done